packages/plugins/users-permissions/server/controllers/auth.js

```diff
@@ -188,7 +188,10 @@ module.exports = {
     }
 
     // Ability to pass OAuth callback dynamically
-    grantConfig[provider].callback = _.get(ctx, 'query.callback') || grantConfig[provider].callback;
+    grantConfig[provider].callback =
+      _.get(ctx, 'query.callback') ||
+      _.get(ctx, 'session.grant.dynamic.callback') ||
```

Comment:
what is this for ?

Reply:
It's a fix for https://github.com/strapi/strapi/issues/7405
The callback query param get lost along the flow and some usecases needs to keep it